### PR TITLE
Add `null` check for `ModelStateEntry.Children`

### PR DESCRIPTION
**Pull request by dougbu**
- #4989

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ValidationHelpers.cs [hunk 1]
```diff
@@ -90,7 +89,7 @@ public static string GetModelErrorMessageOrDefault(ModelError modelError)
             ModelMetadata metadata,
             List<ModelStateEntry> orderedModelStateEntries)
         {
-            if (metadata.ElementMetadata != null)
+            if (metadata.ElementMetadata != null && modelStateEntry.Children != null)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ValidationHelpers.cs]: Unrelated, the `foreach` loop below could be converted to a for loop.

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ValidationHelpers.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ValidationHelpers.cs]: :see_no_evil: 

We have no numbers justifying perf-related changes in this method and this is not a likely hot path.

**Comment by dougbu** [comment]
e5cb6f9

**Pull request closed without merging**